pkg/user/parallel_tx_submission.go

```diff
@@ -0,0 +1,531 @@
+		} else {
+			// Additional workers use generated account names
+			accountName = fmt.Sprintf("parallel-worker-%d", i)
+
+			// Get worker address from keyring if account exists
+			if record, err := client.signer.keys.Key(accountName); err == nil {
+				if addr, err := record.GetAddress(); err == nil {
+					address = addr.String()
+				}
+			}
+		}
+
+		worker := &TxWorker{
+			id:          i,
+			accountName: accountName,
+			address:     address,
+			client:      client,
+			jobQueue:    pool.jobQueue,
+		}
+		pool.workers[i] = worker
+	}
+
+	return pool
+}
+
+// NewParallelTxPool creates a new parallel transaction submission pool.
+func NewParallelTxPool(client *TxClient, numWorkers int, initialize bool) *ParallelTxPool {
+	return newParallelTxPool(client, numWorkers, initialize)
+}
+
+// Start initiates all workers in the pool
+func (p *ParallelTxPool) Start(ctx context.Context) error {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	if p.started.Load() {
+		return nil
+	}
+
+	// Initialize workers if configured to do so
+	if p.initialize && !p.initialized.Load() {
+		if err := p.client.InitializeWorkerAccounts(ctx); err != nil {
+			return fmt.Errorf("failed to initialize worker accounts: %w", err)
+		}
+		p.initialized.Store(true)
+	}
+
+	// Create a new context for this pool instance
+	p.ctx, p.cancel = context.WithCancel(ctx)
+
+	for _, worker := range p.workers {
+		worker.ctx = p.ctx
+		go worker.Start()
+	}
+
+	p.started.Store(true)
+	return nil
+}
+
+// Stop shuts down all workers in the pool
+func (p *ParallelTxPool) Stop() {
+	p.mtx.Lock()
+	defer p.mtx.Unlock()
+
+	if !p.started.Load() {
+		return
+	}
+
+	if p.cancel != nil {
+		p.cancel()
+	}
+	p.started.Store(false)
+}
+
+// SubmitJob submits a job to the parallel worker pool
+func (p *ParallelTxPool) SubmitJob(job *SubmissionJob) {
+	p.mtx.RLock()
+	jobQueue := p.jobQueue
+	ctx := p.ctx
+	p.mtx.RUnlock()
+
+	select {
+	case jobQueue <- job:
+	case <-ctx.Done():
+		job.ResultsC <- SubmissionResult{Error: errors.New("parallel pool full or has stopped")}
+	}
+}
+
+// Workers returns the workers in the parallel pool
+func (p *ParallelTxPool) Workers() []*TxWorker {
+	return p.workers
+}
+
+// IsStarted returns whether the parallel pool is started
+func (p *ParallelTxPool) IsStarted() bool {
+	return p.started.Load()
+}
+
+// Start begins the worker's job processing loop
+func (w *TxWorker) Start() {
+	for {
+		select {
+		case job := <-w.jobQueue:
+			w.processJob(job)
+		case <-w.ctx.Done():
+			return
+		}
+	}
+}
+
+// processJob handles the actual transaction submission for a job
+func (w *TxWorker) processJob(job *SubmissionJob) {
+	jobCtx := job.Ctx
+	if jobCtx == nil {
+		jobCtx = context.Background()
+	}
+
+	select {
+	case <-jobCtx.Done():
+		job.ResultsC <- SubmissionResult{Signer: w.address, Error: jobCtx.Err()}
+		return
+	case <-w.ctx.Done():
+		job.ResultsC <- SubmissionResult{Signer: w.address, Error: w.ctx.Err()}
+		return
+	default:
+	}
```

Comment:
wasn't sure if this is necessary or not, but we will cancel if the ctx passed to SubmitPFB is cancelled or if the total txclient is cancelled.